### PR TITLE
do value update if the value is valid again

### PR DESCRIPTION
**Pull request by halitanildonmez**
## The basics

- [X] I branched from develop
- [X] My pull request is against develop
- [X] My code follows the [style guide](https://developers.google.com/blockly/guides/modify/web/style-guide)

### Resolves

https://github.com/google/blockly/issues/4345

### Proposed Changes

The issue was, in the field, if the user has entered invalid once the flag `isTextValid_;` was being set to false at `doValueInvalid_` which is fine. If however the user enters correct value, then the flag is never se to true.

I made it so that at field.js `processValidation_` we do `doValueUpdate_` so that we update the flag.

### Reason for Changes

To solve the bug that was opened.

### Test Coverage

Tested on:
Desktop Chrome
Desktop Firefox

### CODE REVIEW

**Comment by BeksOmega** [comment]
Hello @halitanildonmez Thanks for submitting a fix for this :D

I think that you're definitely looking in the right spot wrt the bug you mentioned! But I think it might be better to modify the logic of the setValue function, rather than modify the processValidation_ function. With the current fix it might run into problems with doValueUpdate_ and forceRerender getting called on values that pass the class validator, but not the local validator. I don't think this would cause bad behavior or anything, but rerenders can be pretty expensive, so it's best to minimize them when possible. Do you feel like that's workable?

And it seems like the travis build failed because some mocha tests need to be updated for the new behavior. These tests are located in your my-blockly-clone/tests/mocha directory. There is also an index.html file in that directory, and if you open that in your browser it will give you more details about what tests are passing/failing.

Specifically it looks like some sinon assertions need updates, so the [sinon assertions](https://sinonjs.org/releases/latest/assertions/) docs might be helpful.

Thank you again for the submission! I know the fields stuff can be kind of a mess hehe so thanks for digging through it :D If you have any questions/comments obviously please feel free to reply!

**Comment by halitanildonmez** [comment]
Hello @BeksOmega ! 

Thank you for the feedback and the help! :smile: I have fixed it with the last two commits (i hope) and hope it looks better now.

Let me know if you want me to change something else :smile:

**Comment by halitanildonmez** [comment]
one note, I changed the tests to calledOnce for value update, hope that is ok

**Comment by BeksOmega** [comment]
Looks perfect! Thank you for the updates @halitanildonmez :D

**Comment by halitanildonmez** [comment]
Awesome! So glad to hear 😄

File: core/field.js [hunk 1]
```diff
@@ -899,6 +899,7 @@ Blockly.Field.prototype.setValue = function(newValue) {
   var oldValue = this.getValue();
   if (oldValue === newValue) {
     doLogging && console.log('same, return');
```
Comment (alschmiedt) [inline core/field.js:901]: I think this needs to be updated to reflect the change. Something like 'same, doValueUpdate_ then return'.

**Review by alschmiedt — APPROVED** [review]
This looks great! 

Just one comment and then lgtm.

File: core/field.js (hunk 1 above)
Comment (halitanildonmez) [reply core/field.js]: Fixed it! :smile: thank you for the input!

**Pull request merged**